File: ruby/test/wire/review_queue_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class ReviewQueueWireTest < WireMockTestCase
  def setup
    super

    @client = Sikaru::Client.new(
      api_key: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_review_queue_create_review_queue_item_with_wiremock
    test_id = "review_queue.create_review_queue_item.0"

    @client.review_queue.create_review_queue_item(
      project_id: "project_id",
      trace_ids: ["traceIds"],
      request_options: {
        additional_headers: {
          "X-Test-Id" => "review_queue.create_review_queue_item.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/review-queue",
      query_params: nil,
      expected: 1
    )
  end
end

File: rust/tests/review_queue_test.rs
use sikaru::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_review_queue_create_review_queue_item_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = Sikaru::new(config).expect("Failed to build client");

    let result = client
        .review_queue
        .create_review_queue_item(
            &"project_id".to_string(),
            &CreateReviewQueueItemRequest {
                trace_ids: vec!["traceIds".to_string()],
                action: None,
                agent_id: None,
                backfill_window: None,
                filter_expression: None,
                issue_id: None,
                issue_title: None,
                rule_id: None,
                rule_name: None,
                sample_rate: None,
                signal_tags: None,
                target: None,
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("POST", "/v1/projects/project_id/review-queue", None, 1)
        .await
        .unwrap();
}
